src/App.tsx

```diff
@@ -1,10 +1,62 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
+import classNames from 'classnames';
 import { UserWarning } from './UserWarning';
+import { Todo } from './types/Todo';
+import { getTodos } from './api/todos';
+import { TodoList } from './components/TodoList';
+import { Filter } from './components/Filter';
+import { NewTodo } from './components/NewTodo';
+import { FILTERS } from './types/FILTERS';
 
-const USER_ID = 0;
+const USER_ID = 11253;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState<string>('');
+  const [filter, setFilter] = useState(FILTERS.ALL);
+  const visibleTodos = useMemo(() => {
+    switch (filter) {
+      case FILTERS.ALL:
+        return todos;
+
+      case FILTERS.ACTIVE:
+        return todos.filter(todo => !todo.completed);
+
+      case FILTERS.COMPLITED:
+        return todos.filter(todo => todo.completed);
+
+      default:
+        return todos;
+    }
+  }, [todos, filter]);
+
+  useEffect(() => {
+    getTodos(USER_ID)
+      .then(setTodos)
+      .catch();
+  }, []);
+
+  useEffect(() => {
```

Comment:
Using setTimeout in useEffect could lead to memory leaks if the component unmounts before the timeout finishes. Consider using clearTimeout in the cleanup function of useEffect.